cloudevents/work/agent/handler/resourcehandler.go

```diff
@@ -0,0 +1,31 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	workv1client "open-cluster-management.io/api/client/work/clientset/versioned/typed/work/v1"
+	"open-cluster-management.io/api/cloudevents/generic"
+	"open-cluster-management.io/api/cloudevents/generic/types"
+	workv1 "open-cluster-management.io/api/work/v1"
+)
+
+// NewManifestWorkAgentHandler returns a ResourceHandler for a ManifestWork on managed cluster.
+func NewManifestWorkAgentHandler(ctx context.Context, workClient workv1client.ManifestWorkInterface) generic.ResourceHandler[*workv1.ManifestWork] {
+	return func(action types.ResourceAction, work *workv1.ManifestWork) error {
+		switch action {
+		case types.Added:
+			_, err := workClient.Create(ctx, work, metav1.CreateOptions{})
```

Comment:
this is not clear to me, why the resource handler finally creates a manifestwork using a client? I thought you are going to put it in the cache.Store.

Reply:
yes, it will be put in the store, we will have a new client, it implements the `ManifestWorkInterface`, the client will put the manifestwork to the sore, the logic will be same with here, so I used the client to handle this

Reply:
it is confusing I think, if it is a certain type client, we should use that as the input instead of the interface.

Reply:
hrm... I do not introduce the new client in this pr yet, I will add the implementation, and use a certain type client